src/GlobalFormsProvider/GlobalFormsContext.ts

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useContext, useMemo } from 'react';
+
+export type FormInfo = { name: string; order: number };
+type FormInfoMixed = string | FormInfo;
+
+export type CallbackWithOrder = {
+  callback: () => Promise<void>;
+  order: number;
+};
+
+export type GlobalFormsContextType = {
+  isSubmitting: boolean;
+  reset: () => void;
+  setResetCallback: (form: FormInfo, callback: () => void) => void;
+  setSubmitCallback: (form: FormInfo, callback: () => Promise<void>) => void;
+  setSubmitting: (form: FormInfo, submitting: boolean) => void;
+  setSubmitSuccessful: (form: FormInfo, submitSuccessful: boolean) => void;
+  /**
+   * Returns a combined boolean of all submitSuccessful states
+   */
+  submit: (additionalCallback?: CallbackWithOrder) => Promise<boolean>;
+};
+
+export const GlobalFormsContext = React.createContext<GlobalFormsContextType>({
+  isSubmitting: false,
+  reset: () => {},
+  setResetCallback: () => {},
+  setSubmitCallback: () => {},
+  setSubmitting: () => {},
+  setSubmitSuccessful: () => {},
+  submit: () => Promise.resolve(false),
+});
+
+export function useGlobalForms() {
+  const { isSubmitting, reset, submit } = useContext(GlobalFormsContext);
+
+  return {
+    isSubmitting,
+    reset,
+    submit,
+  };
+}
+
+export function normalizeFormInfo(form: FormInfoMixed): FormInfo {
+  if (typeof form === 'string') {
+    return { name: form, order: 0 };
+  }
+
+  return form;
+}
+
+export function useGlobalForm(form: FormInfoMixed) {
+  const formInfo = useMemo(() => normalizeFormInfo(form), [form]);
+
+  const {
+    setResetCallback,
+    setSubmitCallback,
+    setSubmitting,
+    setSubmitSuccessful,
+  } = useContext(GlobalFormsContext);
+
+  const setResetCallbackWrapped = useCallback(
```

Comment:
The wrapper function also cares about passing the `formInfo`, so that when using the callback, less parameters have to be passed. In my opinion the name is good enough. Alternatives might be `setResetCallbackWithoutFormInfo`, `handleSetResetCallback`.